### PR TITLE
Add Sphinx Documentation

### PR DESCRIPTION
**Pull request by chsasank**
Hi,

I've worked on sphinx documentation for SimpleITK. Please see #87 for more details. I'm moving existing documentation files I have from https://github.com/chsasank/sitk-docs to Documentations/docs/ in logical order of commits to make review easy. 

I have written a 'user guide' based on sphinx gallery incorporating code from notebooks and makes tutorials look great. It's not yet complete, partly because I didn't understand some of the notebooks. 
Therefore, you may choose to merge to a branch 'docs' before merging to master.

Todo:

- [x] Move existing docs
- [x] Add sphinx gallery config
- [x] Add Image user guide 
- [x] Add custom directives for gallery item and create user guide page.
- [ ] Add other user guides

More:

- [ ] Update docs for 1.0rc
- [ ] Port rest of the notebooks 
- [ ] Create a way for examples to be shown (with some custom directives and js may be) 
- [ ] Link doxygen documentations

Sasank.

### CODE REVIEW

**Comment by chsasank** [comment]
Following pages are from wiki (with changes improving readability): 
* building/*
* faq
* installation

These may have to be modified for 1.0 rc release.

**Comment by chsasank** [comment]
image guide and visualising image guides are loosely based on following notebooks: 
00 Setup
01 Image Basics
02 Pythonic Image
03 Image Details
10 matplotlib's imshow

I have added a text of my own too and made the text more guide like.

**Comment by chsasank** [comment]
transforms chapter of user guide are based on following notebooks

[complete] interpolation_guide - 20 Expand With Interpolators
[complete] transforms_guide - 21 Transforms and Resampling
[wip] moretransforms_guide - 22 Transforms

If moretransforms_guide is completed, I consider this chapter completed.

**Comment by chsasank** [comment]
For segmentation chapter, this is how it is organised:

threshold_guide - part of 300_Segmentation_Overview
region_growing_guide - 30_Segmentation_Region_Growing

I didn't work on the rest of 3x notebooks.

This is a very long PR. I hope you find this useful.

File: Documentation/docs/source/conf.py [hunk 1]
```diff
@@ -75,6 +80,39 @@
...
+}
+
+# Copy rst in source dirs to build dir
+for i in range(len(sphinx_gallery_conf['examples_dirs'])):
+    gallery_dir = sphinx_gallery_conf['gallery_dirs'][i]
+    source_dir = sphinx_gallery_conf['examples_dirs'][i]
+    # Create gallery dirs if it doesn't exist
+    try:
+        os.mkdir(gallery_dir)
+    except FileExistsError:
```
Comment (dave3d) [inline Documentation/docs/source/conf.py]: Change FileExistsError to OSError so the docs can be built with Python 2.  FileExistsError is only available in Python 3.

**Comment by blowekamp** [comment]
The sphinx documentation has been completed in other patches inspired by this work.

**Pull request closed without merging**